File: app/src/main/java/co/atrasvida/example/ApiClient/BaseResponse.java
package co.atrasvida.example.ApiClient;

import com.google.gson.annotations.SerializedName;

import java.util.List;


class BaseResponse<T> {

    @SerializedName("IsSuccess")
    boolean isSuccess = false;

    @SerializedName("Item")
    T item = null;

    @SerializedName("ListItems")
    List<T> listItems = null;

    @SerializedName("ErrorCode")
    int errorCode = 0;

    @SerializedName("Message")
    String message = null;

    /**
     * this is only for /api/token/ request
     */
    @SerializedName("access")
    String access = null;

}

File: app/src/main/java/co/atrasvida/example/ApiClient/BaseResponse.kt
package co.atrasvida.example.ApiClient

import com.google.gson.annotations.SerializedName


class BaseResponse<T> {

    @SerializedName("IsSuccess")
    var isSuccess: Boolean = false

    @SerializedName("Item")
    var item: T? = null

    @SerializedName("ListItems")
    var listItems: List<T>? = null

    @SerializedName("ErrorCode")
    var errorCode: Int? = null

    @SerializedName("Message")
    var message: String? = null

    /**
     * this is only for /api/token/ request
     */
    @SerializedName("access")
    var access: String? = null

}
